code/CMSBatchAction.php

```diff
@@ -67,16 +67,9 @@ public function response($successMessage, $status)
             }
         }
 
-        $response = Controller::curr()->getResponse();
-
-        if ($response) {
-            $response->setStatusCode(
-                200,
-                sprintf($successMessage ?? '', $count, $errors)
-            );
-        }
-
-        return json_encode($status);
+        return HTTPResponse::create()
+            ->setStatusCode(200, sprintf($successMessage ?? '', $count, $errors))
+            ->setBody(json_encode($status));
```

Comment:
This change will mean any modifications made on the Controller's `HTTPResponse` object won't apply to the back action's response - is that an okay behavioural change to make?

Reply:
Yup that's fine, this is just a helper method used by CMSBatchAction::batchaction() which itself is used by the core implementations of batches actions e.g. `CMSBatchAction_Publish`

These helper methods shouldn't really have been made public in the first place

Reply:
The visibility of the method is irrelevant here imo. You can change a `Controller`'s response from any number of places to change headers, etc.
Though on further reflections I think in this case the old behaviour of fetching the controller and modifying it's response probably was incorrect anyway so no worries.